server/src/main/java/com/cloud/api/dispatch/ParamProcessWorker.java

```diff
@@ -335,13 +337,20 @@ private void setFieldValue(final Field field, final BaseCmd cmdObj, final Object
             case DATE:
                 // This piece of code is for maintaining backward compatibility
                 // and support both the date formats(Bug 9724)
-                final boolean isObjInNewDateFormat = isObjInNewDateFormat(paramObj.toString());
-                if (isObjInNewDateFormat) {
+                try {
+                    field.set(cmdObj, DateUtil.parseTZDateString(paramObj.toString()));
+                    break;
+                } catch (ParseException parseException) {
+                    s_logger.debug(String.format("Could not parse date [%s] with timezone parser, trying to parse without timezone.", paramObj));
+                }
```

Comment:
Can you put this, or even the full case DATE block in a separate method, please_